tests/test_test_scenario.py

```
@@ -922,3 +924,109 @@ def test_nsys_disable_override(self, test_scenario_parser: TestScenarioParser, s
+
+    def test_path_and_test_template_name_together_is_rejected(self):
+        with pytest.raises(ValueError) as exc_info:
+            TestRunModel(id="1", path="nccl.toml", test_template_name="NcclTest")
+        assert exc_info.match("'test_template_name' must not be set if 'test_name' or 'path' is set.")
+
+    def test_path_alone_satisfies_the_base_requirement(self):
+        model = TestRunModel(id="1", path="nccl.toml")
+        assert model.path == "nccl.toml"
+
+    def test_empty_path_is_rejected(self):
+        with pytest.raises(ValueError) as exc_info:
```

Ruff (0.16.1)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)